<!DOCTYPE html>
<html>
  <head>
    <meta charset='utf-8'>
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    <link href='https://fonts.googleapis.com/css?family=Architects+Daughter' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/pygment_trac.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/print.css" media="print">

    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->

    <title>SeqMule by WangGenomicsLab</title>
  </head>

  <body>
    <header>
      <div class="inner">
        <h1>SeqMule</h1>
        <h2>Automated human exome/genome variants detection</h2>
        <a href="https://github.com/WangGenomicsLab/SeqMule" class="button"><small>View project on</small> GitHub</a>
      </div>
    </header>

    <div id="content-wrapper">
      <div class="inner clearfix">
        <section id="main-content">
          <h1>
<a id="seqmule-automated-human-exomegenome-variants-detection" class="anchor" href="#seqmule-automated-human-exomegenome-variants-detection" aria-hidden="true"><span class="octicon octicon-link"></span></a>SeqMule: Automated human exome/genome variants detection</h1>

<p>SeqMule takes single-end or paird-end FASTQ or BAM files, generates a script consisting of more than 10 popular alignment, analysis tools and runs the script line by line. Users can change the pipeline or fine-tune the parameters by modifying its configuration file. SeqMule also has some built-in functions, such as pooling consensus calls from various callers, plotting a Venn diagram showing intersection among different callers, and downloading databases.</p>

<h2>
<a id="synopsis" class="anchor" href="#synopsis" aria-hidden="true"><span class="octicon octicon-link"></span></a>Synopsis</h2>

<ul>
<li><p><strong>seqmule download</strong>: download databases/BEDs that are required by sequence alignment or variant calling software tools</p></li>
<li><p><strong>seqmule pipeline</strong>: perform the automated pipeline for detection of variants from whole-exome/genome data</p></li>
<li><p><strong>seqmule stats</strong>: perform statistical analysis of variants data, such as drawing Venn diagram to examine overlap between VCF files, generating union/consensus ca
lls, generating coverage/alignment statistics in specific genomic regions, calculating Mendelian error rates</p></li>
<li><p><strong>seqmule run</strong>: continue run from last executed step after interruption or run from a specific step</p></li>
<li><p><strong>seqmule update</strong>: perform automated update of the SeqMule software tools</p></li>
</ul>

<h2>
<a id="features" class="anchor" href="#features" aria-hidden="true"><span class="octicon octicon-link"></span></a>Features</h2>

<ul>
<li>
<p>Multiple aligners</p>

<p>BWA-MEM, BWA-BACKTRACK, Bowtie, Bowtie2, SOAP2, SNAP</p>
</li>
<li>
<p>Multiple variant callers</p>

<p>GATK, SAMtools, VarScan, SOAPsnp, Freebayes are available.</p>

<p>As stated on 1000 Genomes Project website, genotypes obtained through a consensus procedure are estimated to have 30% fewer errors than those generated by any single caller. </p>

<p>As we have demonstrated in a previous study (O'Rawe et al. <em>Genome Med</em> 2013, <strong>5</strong>:28), consensus calls from multiple calling algorithms may increase calling accuracy and reduce Mendelian error rates.</p>
</li>
<li>
<p>Easy downloading and installation.</p>

<p>Most jobs can be done with one-line command.</p>
</li>
<li>
<p>Fast and easy customization</p>

<p>Just change the 'advanced_config' file!</p>
</li>
</ul>

<h2>
<a id="dependencies" class="anchor" href="#dependencies" aria-hidden="true"><span class="octicon octicon-link"></span></a>Dependencies</h2>

<ul>
<li>R</li>
<li>Perl</li>
<li>Java</li>
</ul>

<h2>
<a id="more-information" class="anchor" href="#more-information" aria-hidden="true"><span class="octicon octicon-link"></span></a>More information</h2>

<ul>
<li><p><a href="http://seqmule.usc.edu">SeqMule Homepage</a></p></li>
<li><p><a href="http://genomics.usc.edu">Wang Genomics Lab Homepage</a></p></li>
</ul>
        </section>

        <aside id="sidebar">
          <a href="https://github.com/WangGenomicsLab/SeqMule/zipball/master" class="button">
            <small>Download</small>
            .zip file
          </a>
          <a href="https://github.com/WangGenomicsLab/SeqMule/tarball/master" class="button">
            <small>Download</small>
            .tar.gz file
          </a>

          <p class="repo-owner"><a href="https://github.com/WangGenomicsLab/SeqMule"></a> is maintained by <a href="https://github.com/WangGenomicsLab">WangGenomicsLab</a>.</p>

          <p>This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the Architect theme by <a href="https://twitter.com/jasonlong">Jason Long</a>.</p>
        </aside>
      </div>
    </div>

            <script type="text/javascript">
            var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
            document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
          </script>
          <script type="text/javascript">
            try {
              var pageTracker = _gat._getTracker("UA-58410868-1");
            pageTracker._trackPageview();
            } catch(err) {}
          </script>

  </body>
</html>
