<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<title>SeqMule Report</title>
<meta name="keywords" content="green jelly, theme, free templates, nivo image slider, website, templatemo, CSS, HTML" />
<meta name="description" content="Green Jelly Theme, free CSS template provided by templatemo.com" />
<link href="../css/templatemo_style.css" rel="stylesheet" type="text/css" />

<link rel="stylesheet" href="../css/nivo-slider.css" type="text/css" media="screen" />

<script language="javascript" type="text/javascript">
function clearText(field)
{
    if (field.defaultValue == field.value) field.value = '';
    else if (field.value == '') field.value = field.defaultValue;
}
</script>

<link rel="stylesheet" type="text/css" href="../css/ddsmoothmenu.css" />

<script type="text/javascript" src="../js/jquery.min.js"></script>
<script type="text/javascript" src="../js/ddsmoothmenu.js">

/***********************************************
* Smooth Navigational Menu- (c) Dynamic Drive DHTML code library (www.dynamicdrive.com)
* This notice MUST stay intact for legal use
* Visit Dynamic Drive at http://www.dynamicdrive.com/ for full source code
***********************************************/

</script>

<script type="text/javascript">

ddsmoothmenu.init({
	mainmenuid: "templatemo_menu", //menu DIV id
	orientation: 'h', //Horizontal or vertical menu: Set to "h" or "v"
	classname: 'ddsmoothmenu', //class added to menu's outer DIV
	//customtheme: ["#1c5a80", "#18374a"],
	contentsource: "markup" //"markup" or ["container_id", "path_to_menu_file"]
})

</script>
  
</head>
<body>

<div id="templatemo_wrapper">
	<div id="templatemo_header"></div> <!-- end of header -->
      <div id="templatemo_menu" class="ddsmoothmenu">
        <ul>
            <li><a href="../summary.html" >Summary</a></li>
            <li><a href="#">Samples</a>
<ul>
<li><a href='../father/father.html'>father</a></li>
<li><a href='../mother/mother.html'>mother</a></li>
<li><a href='../son/son.html'>son</a></li>
<span></span>
</ul>
<ul>
<li><a href='../father/father.html'>father</a></li>
<li><a href='../mother/mother.html'>mother</a></li>
<li><a href='../son/son.html'>son</a></li>
<span></span>
</ul>
            </li>
            <li><a href="pipeline.html">Pipeline</a> </li>
          	<li><a href="parameter.html" class="selected">Parameter</a></li>
          	<li><a href="http://seqmule.usc.edu">Help</a></li>
            </ul>
        <br style="clear: left" />
    </div> <!-- end of menu -->
    
    <div id="templatemo_slider_wrapper">
      <div id="htmlcaption" class="nivo-html-caption">
       	  <strong>This</strong> is an example of a HTML caption with <a href="#">a link</a>.
        </div>
    
    </div>
    
    <script type="text/javascript" src="../js/jquery-1.4.3.min.js"></script>
    <script type="text/javascript" src="../js/jquery.nivo.slider.js"></script>
    <script type="text/javascript">
    $(window).load(function() {
    $('#slider').nivoSlider();
    });
    </script>
    
    <div id="templatemo_main"><br class="cleaner" />
<h1>Analysis Steps and Parameters</h1>
<p>DISABLED steps are NOT shown here.</p>
<h2>FastQC(version: 0.10.1)</h2>
<h2>BWA-MEM(version: 0.7.8-r455)</h2>
<table><tr><th>Options</th><th>Values</th></tr>
<tr>
			<td>gapOpenPenalty</td>
			<td>6</td>
			</tr>
<tr>
			<td>clipPenalty</td>
			<td>5</td>
			</tr>
<tr>
			<td>additional</td>
			<td></td>
			</tr>
<tr>
			<td>gapExtensionPenalty</td>
			<td>1</td>
			</tr>
<tr>
			<td>matchScore</td>
			<td>1</td>
			</tr>
<tr>
			<td>readSingletonPenalty</td>
			<td>17</td>
			</tr>
<tr>
			<td>mismatchPenalty</td>
			<td>4</td>
			</tr>
</table><br><br>
<h2>Picard(subprogram: markduplicates; version: 1.112)</h2>
<table><tr><th>Options</th><th>Values</th></tr>
<tr>
			<td>forceRm</td>
			<td>0</td>
			</tr>
</table><br><br>
<h2>SAMtools(subprogram: filter; version: 0.1.19-44428cd)</h2>
<table><tr><th>Options</th><th>Values</th></tr>
<tr>
			<td>mapQ</td>
			<td>30</td>
			</tr>
</table><br><br>
<h2>SAMtools(subprogram: index; version: 0.1.19-44428cd)</h2>
<h2>GATK-Full(subprogram: realign; version: 3.1-1-g07a4bf8)</h2>
<table><tr><th>Options</th><th>Values</th></tr>
<tr>
			<td>maxInMemory</td>
			<td>200000</td>
			</tr>
<tr>
			<td>additional</td>
			<td> -rf NotPrimaryAlignment --consensusDeterminationModel USE_READS -l ERROR</td>
			</tr>
<tr>
			<td>LOD</td>
			<td>0.4</td>
			</tr>
</table><br><br>
<h2>GATK-Full(subprogram: recal; version: 3.1-1-g07a4bf8)</h2>
<table><tr><th>Options</th><th>Values</th></tr>
<tr>
			<td>additional</td>
			<td>--disable_indel_quals -rf BadCigar -l ERROR</td>
			</tr>
<tr>
			<td>covariates</td>
			<td>ReadGroupCovariate,QualityScoreCovariate,ContextCovariate,CycleCovariate</td>
			</tr>
</table><br><br>
<h2>GATK-Full(subprogram: HaplotypeCaller; version: 3.1-1-g07a4bf8)</h2>
<table><tr><th>Options</th><th>Values</th></tr>
<tr>
			<td>numBadVariants</td>
			<td>5000</td>
			</tr>
<tr>
			<td>forceINDELVQSR</td>
			<td>0</td>
			</tr>
<tr>
			<td>indelMaxGaussians</td>
			<td>4</td>
			</tr>
<tr>
			<td>snpfilters</td>
			<td>QDFilter,QD<2.0 FSFilter,FS>60.0;MQFilter,MQ<40.0;MaqQualRankSumFilter,MappingQualityRankSum<-12.5;ReadPosFilter,ReadPosRankSum<-8.0</td>
			</tr>
<tr>
			<td>originalBAM</td>
			<td>0</td>
			</tr>
<tr>
			<td>HCadditional</td>
			<td>  -rf BadCigar -l ERROR</td>
			</tr>
<tr>
			<td>maxGaussians</td>
			<td>4</td>
			</tr>
<tr>
			<td>filter</td>
			<td>1</td>
			</tr>
<tr>
			<td>indeltsfilter</td>
			<td>99.0</td>
			</tr>
<tr>
			<td>forceSNPVQSR</td>
			<td>0</td>
			</tr>
<tr>
			<td>indelfilters</td>
			<td>QDFilter,QD<2.0 ReadPosFilter,ReadPosRankSum<-20.0;FSFilter,FS>200.0</td>
			</tr>
<tr>
			<td>annotations</td>
			<td>QD,MQRankSum,ReadPosRankSum,FS,MQ</td>
			</tr>
<tr>
			<td>indelNumBadVariants</td>
			<td>1000</td>
			</tr>
<tr>
			<td>indelAnnotations</td>
			<td>DP,FS,MQRankSum,ReadPosRankSum</td>
			</tr>
<tr>
			<td>snptsfilter</td>
			<td>99.0</td>
			</tr>
<tr>
			<td>clusterWindowSize</td>
			<td>10</td>
			</tr>
</table><br><br>
<h2>SAMtools(subprogram: mpileup; version: 0.1.19-44428cd)</h2>
<table><tr><th>Options</th><th>Values</th></tr>
<tr>
			<td>minDepth</td>
			<td>10</td>
			</tr>
<tr>
			<td>minMapQBias</td>
			<td>0</td>
			</tr>
<tr>
			<td>FILTERadditional</td>
			<td></td>
			</tr>
<tr>
			<td>BCFadditional</td>
			<td></td>
			</tr>
<tr>
			<td>minStrandBias</td>
			<td>0.0001</td>
			</tr>
<tr>
			<td>bwaDownGrade</td>
			<td>50</td>
			</tr>
<tr>
			<td>minEndDistBias</td>
			<td>0.0001</td>
			</tr>
<tr>
			<td>minBaseQBias</td>
			<td>1e-100</td>
			</tr>
<tr>
			<td>SAMadditional</td>
			<td></td>
			</tr>
<tr>
			<td>originalBAM</td>
			<td>1</td>
			</tr>
</table><br><br>
<h2>VarScan(version: 2.3)</h2>
<table><tr><th>Options</th><th>Values</th></tr>
<tr>
			<td>minDepth</td>
			<td>10</td>
			</tr>
<tr>
			<td>VARadditional</td>
			<td></td>
			</tr>
<tr>
			<td>minBaseQ</td>
			<td>20</td>
			</tr>
<tr>
			<td>P</td>
			<td>0.1</td>
			</tr>
<tr>
			<td>minSupport</td>
			<td>2</td>
			</tr>
<tr>
			<td>SAMadditional</td>
			<td></td>
			</tr>
<tr>
			<td>originalBAM</td>
			<td>1</td>
			</tr>
</table><br><br>

    </div> <!-- end of main -->
    
    <div id="templatemo_footer">
      <div class="cleaner"></div>
    </div> <!-- end of footer -->
</div>

<div id="templatemo_cr_bar_wrapper">
	<div id="templatemo_cr_bar">
    	Copyright 2014 <a href="http://genomics.usc.edu">USC Wang Lab</a> | Designed by <a href="http://www.templatemo.com" target="_parent">Free CSS Templates</a>
    </div>
</div>


<script type='text/javascript' src='../js/logging.js'></script>
</body>
</html>
