<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<title>SeqMule Report</title>
<meta name="keywords" content="green jelly, theme, free templates, nivo image slider, website, templatemo, CSS, HTML" />
<meta name="description" content="Green Jelly Theme, free CSS template provided by templatemo.com" />
<link href="../css/templatemo_style.css" rel="stylesheet" type="text/css" />

<link rel="stylesheet" href="../css/nivo-slider.css" type="text/css" media="screen" />

<script language="javascript" type="text/javascript">
function clearText(field)
{
    if (field.defaultValue == field.value) field.value = '';
    else if (field.value == '') field.value = field.defaultValue;
}
</script>

<link rel="stylesheet" type="text/css" href="../css/ddsmoothmenu.css" />

<script type="text/javascript" src="../js/jquery.min.js"></script>
<script type="text/javascript" src="../js/ddsmoothmenu.js">

/***********************************************
* Smooth Navigational Menu- (c) Dynamic Drive DHTML code library (www.dynamicdrive.com)
* This notice MUST stay intact for legal use
* Visit Dynamic Drive at http://www.dynamicdrive.com/ for full source code
***********************************************/

</script>

<script type="text/javascript">

ddsmoothmenu.init({
	mainmenuid: "templatemo_menu", //menu DIV id
	orientation: 'h', //Horizontal or vertical menu: Set to "h" or "v"
	classname: 'ddsmoothmenu', //class added to menu's outer DIV
	//customtheme: ["#1c5a80", "#18374a"],
	contentsource: "markup" //"markup" or ["container_id", "path_to_menu_file"]
})

</script>
  
</head>
<body>

<div id="templatemo_wrapper">
	<div id="templatemo_header"></div> <!-- end of header -->
      <div id="templatemo_menu" class="ddsmoothmenu">
        <ul>
            <li><a href="../summary.html">Summary</a></li>
            <li><a href="#">Samples</a>
            </li>
            <li><a href="pipeline.html" class="selected">Pipeline</a> </li>
          	<li><a href="parameter.html">Parameter</a></li>
          	<li><a href="help.html">Help</a></li>
            </ul>
        <br style="clear: left" />
    </div> <!-- end of menu -->
    
    <div id="templatemo_slider_wrapper">
      <div id="htmlcaption" class="nivo-html-caption">
       	  <strong>This</strong> is an example of a HTML caption with <a href="#">a link</a>.
        </div>
    
    </div>
    
    <script type="text/javascript" src="../js/jquery-1.4.3.min.js"></script>
    <script type="text/javascript" src="../js/jquery.nivo.slider.js"></script>
    <script type="text/javascript">
    $(window).load(function() {
    $('#slider').nivoSlider();
    });
    </script>
    
    <div id="templatemo_main"><br class="cleaner" />


	    <h1>Analysis Pipeline</h1>
	    <h2>FASTQ input</h2>
	    <img class='center' src="../images/seqmule_workflow_fq.png" width=506 height=497>
	    <br /><br />
	    <p>
If input is FASTQ files, they will first be checked by FastQC to ensure sequencing quality. Then they are subjected to alignment., and we get BAM files  after this step.

Depending on the length of capture region, SeqMule will try to remove possible PCR and optical duplicates with PicardTools.  Next one or more variant callers will be used to call variants. If we get multiple variant files, SeqMule will automatically extract consensus calls.
	    </p>

	    <h2>BAM input</h2>
	    <img class='center' src="../images/seqmule_workflow_bam.png" width=506 height=497>
	    <br /><br />
	    <p>
If input is BAM files, they will first be converted to FASTQ files, and then checked by FastQC to ensure sequencing quality. The rationale for this conversion is that it is hard to tell which reference genome was used for alignment. Then they are subjected to alignment., and we get BAM files  after this step.

Depending on the length of capture region, SeqMule will try to remove possible PCR and optical duplicates with PicardTools.  Next one or more variant callers will be used to call variants. If we get multiple variant files, SeqMule will automatically extract consensus calls.
	    </p>


    </div> <!-- end of main -->
    
    <div id="templatemo_footer">
      <div class="cleaner"></div>
    </div> <!-- end of footer -->
</div>

<div id="templatemo_cr_bar_wrapper">
	<div id="templatemo_cr_bar">
    	Copyright 2014 <a href="http://genomics.usc.edu">USC Wang Lab</a> | Designed by <a href="http://www.templatemo.com" target="_parent">Free CSS Templates</a>
    </div>
</div>


<script type='text/javascript' src='../js/logging.js'></script>
</body>
</html>
