<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<title>SeqMule Report</title>
<meta name="keywords" content="green jelly, theme, free templates, nivo image slider, website, templatemo, CSS, HTML" />
<meta name="description" content="Green Jelly Theme, free CSS template provided by templatemo.com" />
<link href="../css/templatemo_style.css" rel="stylesheet" type="text/css" />

<link rel="stylesheet" href="../css/nivo-slider.css" type="text/css" media="screen" />

<script language="javascript" type="text/javascript">
function clearText(field)
{
    if (field.defaultValue == field.value) field.value = '';
    else if (field.value == '') field.value = field.defaultValue;
}
</script>

<link rel="stylesheet" type="text/css" href="../css/ddsmoothmenu.css" />

<script type="text/javascript" src="../js/jquery.min.js"></script>
<script type="text/javascript" src="../js/ddsmoothmenu.js">

/***********************************************
* Smooth Navigational Menu- (c) Dynamic Drive DHTML code library (www.dynamicdrive.com)
* This notice MUST stay intact for legal use
* Visit Dynamic Drive at http://www.dynamicdrive.com/ for full source code
***********************************************/

</script>

<script type="text/javascript">

ddsmoothmenu.init({
	mainmenuid: "templatemo_menu", //menu DIV id
	orientation: 'h', //Horizontal or vertical menu: Set to "h" or "v"
	classname: 'ddsmoothmenu', //class added to menu's outer DIV
	//customtheme: ["#1c5a80", "#18374a"],
	contentsource: "markup" //"markup" or ["container_id", "path_to_menu_file"]
})

</script>
  
</head>
<body>

<div id="templatemo_wrapper">
	<div id="templatemo_header"></div> <!-- end of header -->
      <div id="templatemo_menu" class="ddsmoothmenu">
        <ul>
            <li><a href="../summary.html" >Summary</a></li>
            <li><a href="#" class="selected">Samples</a>
<ul>
<li><a href='father.html'>father</a></li>
<li><a href='../mother/mother.html'>mother</a></li>
<span></span>
</ul>
            </li>
            <li><a href="../pages/pipeline.html">Pipeline</a> </li>
          	<li><a href="../pages/parameter.html">Parameter</a></li>
          	<li><a href="../pages/help.html">Help</a></li>
            </ul>
        <br style="clear: left" />
    </div> <!-- end of menu -->
    
    <script type="text/javascript" src="../js/jquery-1.4.3.min.js"></script>
    <script type="text/javascript" src="../js/jquery.nivo.slider.js"></script>
    <script type="text/javascript">
    $(window).load(function() {
    $('#slider').nivoSlider();
    });
    </script>
    
    <div id="templatemo_main"><br class="cleaner" />
<h1>father</h1>
<h2>Input Quality Assessment</h2>
<br><a href='father.0.0_phred33_fastqc/fastqc_report.html'>father.0.0_phred33</a><br><a href='father.0.1_phred33_fastqc/fastqc_report.html'>father.0.1_phred33</a><br><a href='father.1.0_phred33_fastqc/fastqc_report.html'>father.1.0_phred33</a><br><a href='father.1.1_phred33_fastqc/fastqc_report.html'>father.1.1_phred33</a><br><br>
<h2>Statistics</h2>
<table><caption>Alignment stats</caption>
<tr><td colspan=2>source: father_result/father_bwamem.merge.bam</td></tr>
<tr><td>QC-passed reads</td><td>3992668</td></tr>
<tr><td>QC-failed-reads</td><td>0</td></tr>
<tr><td>Duplicates</td><td>0</td></tr>
<tr><td>Mapped reads</td><td>3575179(89.54%)</td></tr>
<tr><td>Paired reads</td><td>3992668</td></tr>
<tr><td>Read1</td><td>1996332</td></tr>
<tr><td>Read2</td><td>1996336</td></tr>
<tr><td>Properly paired</td><td>3914772(98.05%)</td></tr>
<tr><td>Reads with itself and mate mapped</td><td>3574162</td></tr>
<tr><td>Singletons</td><td>1017(0.03%)</td></tr>
<tr><td>Reads with mate mapped to a different chromosome</td><td>21721</td></tr>
<tr><td>Reads with mate mapped to a different chr (mapQ at least 5)</td><td>21721</td></tr>
</table><br><br>
<table><caption>Coverage stats</caption>
<tr><td colspan=2>source: father_result/father_bwamem.merge.bam</td></tr>
<tr><td>total length (defined by /auto/rcf-proj/kw/yunfeigu/Downloads/SeqMule/misc/hg19_exome.bed) </td><td> 46.21Mb</td></tr>
<tr><td>Fraction of reads mapped to target region </td><td> 53.28%</td></tr>
<tr><td>Average coverage in target region </td><td> 3.66</td></tr>
<tr><td>Percentage above 30</td><td>0.12%</td></tr>
<tr><td>Percentage above 20</td><td>0.77%</td></tr>
<tr><td>Percentage above 10</td><td>7.78%</td></tr>
<tr><td>Percentage above 5</td><td>30.53%</td></tr>
</table><br><br>
<table><caption>Variant stats(filter applied)</caption>
<tr><td colspan=2>source: father_result/father_bwamem.merge.0_samtools.extract.vcf</td></tr>
<tr><td>Sample</td><td>father</td></tr>
<tr><td>Number of variants</td><td>3269</td></tr>
<tr><td>Number of SNVs</td><td>3203</td></tr>
<tr><td>Number of indels</td><td>66</td></tr>
<tr><td>Transitions</td><td>2196</td></tr>
<tr><td>Transversions</td><td>1007</td></tr>
<tr><td>Ti/Tv Ratio</td><td>2.18</td></tr>
<tr><td>Ref/Alt heterozygotes</td><td>2389</td></tr>
<tr><td>Homozygotes</td><td>880</td></tr>
</table><br><br>

<h2>Coverage Plot</h2>
<img class='center' src='../father/father_cov.jpg' width=500 height=1000>


    </div> <!-- end of main -->
    
    <div id="templatemo_footer">
      <div class="cleaner"></div>
    </div> <!-- end of footer -->
</div>

<div id="templatemo_cr_bar_wrapper">
	<div id="templatemo_cr_bar">
    	Copyright 2014 <a href="http://genomics.usc.edu">USC Wang Lab</a> | Designed by <a href="http://www.templatemo.com" target="_parent">Free CSS Templates</a>
    </div>
</div>


<script type='text/javascript' src='../js/logging.js'></script>
</body>
</html>
