<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<title>SeqMule Report</title>
<meta name="keywords" content="green jelly, theme, free templates, nivo image slider, website, templatemo, CSS, HTML" />
<meta name="description" content="Green Jelly Theme, free CSS template provided by templatemo.com" />
<link href="../css/templatemo_style.css" rel="stylesheet" type="text/css" />

<link rel="stylesheet" href="../css/nivo-slider.css" type="text/css" media="screen" />

<script language="javascript" type="text/javascript">
function clearText(field)
{
    if (field.defaultValue == field.value) field.value = '';
    else if (field.value == '') field.value = field.defaultValue;
}
</script>

<link rel="stylesheet" type="text/css" href="../css/ddsmoothmenu.css" />

<script type="text/javascript" src="../js/jquery.min.js"></script>
<script type="text/javascript" src="../js/ddsmoothmenu.js">

/***********************************************
 * Smooth Navigational Menu- (c) Dynamic Drive DHTML code library (www.dynamicdrive.com)
 * This notice MUST stay intact for legal use
 * Visit Dynamic Drive at http://www.dynamicdrive.com/ for full source code
 ***********************************************/

</script>

<script type="text/javascript">

ddsmoothmenu.init({
mainmenuid: "templatemo_menu", //menu DIV id
orientation: 'h', //Horizontal or vertical menu: Set to "h" or "v"
classname: 'ddsmoothmenu', //class added to menu's outer DIV
//customtheme: ["#1c5a80", "#18374a"],
contentsource: "markup" //"markup" or ["container_id", "path_to_menu_file"]
})

</script>

</head>
<body>

<div id="templatemo_wrapper">
    <div id="templatemo_header"></div> <!-- end of header -->
    <div id="templatemo_menu" class="ddsmoothmenu">
	<ul>
	    <li><a href="../home.html" >Home</a></li>
	    <li><a href="installation.html">Install</a> </li>
	    <li><a href="tutorial.html" class="selected">Tutorial</a> </li>
	    <li><a href="manual.html">Manual</a></li>
	    <li><a href="faq.html">FAQ</a></li>
	    <li><a href="technotes.html">TechNotes</a></li>
	</ul>
	<br style="clear: left" />
    </div> <!-- end of menu -->

    <script type="text/javascript" src="../js/jquery-1.4.3.min.js"></script>
    <script type="text/javascript" src="../js/jquery.nivo.slider.js"></script>
    <script type="text/javascript">
$(window).load(function() {
	$('#slider').nivoSlider();
	});
    </script>

    <div id="templatemo_main"><br class="cleaner" />
	<h2>Tutorial</h2>
	<p>After you have downloaded and installed SeqMule (assume in <b>seqmule</b> folder), this tutorial will tell you most important steps to get your analysis done.</p>
	<br>
	<div id="table-of-content"></div>

	<h3 id="q1">QUICK START</h3>
	<h4>Step 1: Prepare database</h4>
	<p>Reference genome, index of reference genome, various SNP and INDEL databases are needed for alignment and variant calling. The following command will download all necessary files.</p>
	<br><code>seqmule download -down hg19all<br></code>
	<br>
	<h4>Step 2: Prepare input</h4>
	<p>Once external databases are downloaded. SeqMule is ready for analysis! If you do not have data yet, please download the following example files (<a href="../example/example.1.fq.gz">example.1.fq.gz</a>,<a href="../example/example.2.fq.gz">example.2.fq.gz</a>).</p>
	<br><code>seqmule download -down hg19all<br></code>
	<br>
	<h4>Step 3: Run pipeline</h4>
	<br>
	<code>
	seqmule pipeline -a example.1.fq.gz -b example.2.fq.gz -prefix exomeData -capture default -threads 4 -e<br>
	</code>
	<p><b>example.1.fq.gz</b> and <b>example.2.fq.gz</b> are the FASTQ files you get from a sequencer, in gzipped format. They contain reads and read qualities. Assuming you did paired-end sequencing, there are two files. <b>-prefix exomeData</b> tells SeqMule your sample name is 'exomeData'. <b>-capture default</b> asks SeqMule to use default region definition file.<b>-threads 4</b> asks SeqMule to use 4 threads wherever possible. Change it if you don't have 4 CPUs to use. <b>-e</b> means this data set is exome or captured sequencing data set (not whole genome data set). The default analysis pipeline conists of BWA-MEM, GATKLite and SAMtools. </p>
	<br>
	<h4>Step 4: Check results</h4>
	<p>Wait a few hours until all executions are finished. In the directory where you run your analysis, <b>exomeData_report</b> contains a report in HTML format (webpage), <b>exomeData_result</b> contains alignment results (in BAM format) and variants (in VCF format). Download the report folder as a whole to your computer, open <b>Summary.html</b> with any browser to view summary statistics about your analysis.</p>
	<br>
	    <br><br>
	<h3 id="q2">DATABASE PREPARATION</h3>
	<p>SeqMule requires external databases to work. You can either use your own databases, or download default databases using the following commands. Refer to manual of a specific tool to figure out what database exactly is needed. We recommend you use DEFAULT databases due to software compatibility issues. All databases will be downloaded to <b>seqmule/database</b> directory. Right now only human genome is supported.</p>
	<br>
	<p>The following commands download default databases to seqmule directory for hg19(GRch37) and hg18(GRch36) genome build, respecitvely. The default location for storing databases is <b>seqmule/database</b>.</p>
	<br>
	<code>
	    seqmule download -down hg19all<br>
	    seqmule download -down hg18all</code>
	<br><p>All region definition (BED) files are also downloaded along with databases and saved inside individual folders with manufacture name.</p>
	    <br>
	<h3 id="q3">USE REGION DEFINITION (BED) FILE</h3>
	<br>
	<p>If you have downloaded region definition files for different capture kits, you can use them in your analysis. By default they are located inside <b>seqmule/database</b>. Find your region definition file according to manufacture and capture kit. Then use <b>-capture path_to_your_file</b> along with other options for pipeline command. For example</p>
	<br>
	<code>
	    <b>seqmule pipeline -a example.1.fq.gz -b example.2.fq.gz -prefix exomeData -capture <strong>seqmule/database/hg19agilent/hg19_SureSelect_Human_All_Exon_V5.bed</strong> -threads 4 -e</b><br>
	</code>
	<p>Each line in BED represents a region in your captured sequence. Definition of BED format can be found on <a href='http://genome.ucsc.edu/FAQ/FAQformat.html#format1'>UCSC genome browser.</a></p>
	<br>
	<h3 id="q4">CHANGE PIPELINE</h3>
	<br>
	<p>If you want to change the default pipeline:</p>
	<code>
	    <b>seqmule pipeline -advanced</b><br>
	</code>
	<p>This will generate a copy of 'advanced_config'. This configuration file specifies every step of the pipeline. If you want to add or remove a program from the pipeline, just navigate to that line, change the bit '0' or '1' after the equal sign (=). For example, if you want to use bowtie2 instead of bwa-mem for alignment, change the following lines:</p>
	<br>
	<code>
	    2P_bwamem=1<br>
	    2P_bowtie2=0
	</code>
	<p>To:</p>
	<code>
	    2P_bwamem=0<br>
	    2P_bowtie2=1
	</code>
	<p>Some lines between them are not shown here. Comments begin with '#', they have no effects on other parts of the file. Read the comments at the beginning of file for detailed instructions on how to modify it.</p>
	<p>After you have modified the 'advanced_config' file, you can use it by prepend <b>-advanced advanced_config</b> to your analysis command.</p>
	<code>
	    <b>seqmule pipeline -a 1.fastq -b 2.fastq -prefix exomeData -capture default -threads 4 -e -advanced advanced_config</b><br>
	</code>
	<p>Some users maybe don't know how to edit a file on Linux if they don't have graphics user interface (GUI). The easiest way is download it to your PC, change it with NotePad, upload it. Alternatively, you can refer to a VIM tutorial (it's everywhere on the web!).</p>
	<br>
	<h3 id="q5">WHOLE GENOME ANALYSIS</h3>
	<br>
	<p>Change <b>-e</b> to <b>-g</b> in seqmule command.</p>
	<br>
	<h3 id="q6">MULTIPLE SAMPLES</h3>
	<br>
	<p>Suppose you have two samples, sampleA and sampleB, how to perform basic analysis? (<b>.fq</b> is the same thing as <b>.fastq</b>)</p>
	<code>
	    <b>seqmule pipeline -a sampleA.1.fq,sampleB.1.fq -b sampleA.2.fq,sampleB.2.fq -e -prefix sampleA,sampleB</b><br>
	</code>
	<p>Basically you concatenate input files or sample names by commas. If you know your samples come from a single family, you can do multi-sample variant calling by simply adding <b>-ms</b> to the above command. It is believed that multi-sample calling could be more accurate.</p>
	<br>
	<h3 id="q7">EXAMPLE OUTPUT</h3>
	<br>
	<p><a href="../example_report/summary.html">Example output</a></p>
	<p>You can find an application example in my (<A HREF="../images/SeqMule-ASHG-2012.pdf">poster</A>) at 2012 ASHG meeting.</p>
	<br>
	<br>
	<h3 id="q8">CAVEAT</h3>
	<br>
	<p>
	NOT all combinations of alingers and variant callers work. For example, SOAPaligner and SOAPsnp don't support SAM, BAM formats natively, so they don't work well with the rest of algorithms. Also, bowtie doesn't report mapping quality, so it shouldn't be used with GATK. For combinations we have tested, please use predefined configuration files under <b>seqmule/misc/predefined_config</b> folder.<br>
	</p> <br>
    </div> <!-- end of main -->

    <div id="templatemo_footer">
	<div class="cleaner"></div>
    </div> <!-- end of footer -->
</div>

<div id="templatemo_cr_bar_wrapper">
    <div id="templatemo_cr_bar">
	Copyright 2014 <a href="http://genomics.usc.edu">USC Wang Lab</a> | Designed by <a href="http://www.templatemo.com" target="_parent">Free CSS Templates</a>
    </div>
</div>


<script src='../js/custom.js'></script>
<script type='text/javascript' src='../js/logging.js'></script>
</body>
</html>
