<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<title>SeqMule Report</title>
<meta name="keywords" content="green jelly, theme, free templates, nivo image slider, website, templatemo, CSS, HTML" />
<meta name="description" content="Green Jelly Theme, free CSS template provided by templatemo.com" />
<link href="css/templatemo_style.css" rel="stylesheet" type="text/css" />

<link rel="stylesheet" href="css/nivo-slider.css" type="text/css" media="screen" />

<script language="javascript" type="text/javascript">
function clearText(field)
{
    if (field.defaultValue == field.value) field.value = '';
    else if (field.value == '') field.value = field.defaultValue;
}
</script>

<link rel="stylesheet" type="text/css" href="css/ddsmoothmenu.css" />

<script type="text/javascript" src="js/jquery.min.js"></script>
<script type="text/javascript" src="js/ddsmoothmenu.js">

/***********************************************
* Smooth Navigational Menu- (c) Dynamic Drive DHTML code library (www.dynamicdrive.com)
* This notice MUST stay intact for legal use
* Visit Dynamic Drive at http://www.dynamicdrive.com/ for full source code
***********************************************/

</script>

<script type="text/javascript">

ddsmoothmenu.init({
	mainmenuid: "templatemo_menu", //menu DIV id
	orientation: 'h', //Horizontal or vertical menu: Set to "h" or "v"
	classname: 'ddsmoothmenu', //class added to menu's outer DIV
	//customtheme: ["#1c5a80", "#18374a"],
	contentsource: "markup" //"markup" or ["container_id", "path_to_menu_file"]
})

</script>
  
</head>
<body>

<div id="templatemo_wrapper">
	<div id="templatemo_header"></div> <!-- end of header -->
      <div id="templatemo_menu" class="ddsmoothmenu">
        <ul>
            <li><a href="summary.html" class="selected">Summary</a></li>
            <li><a href="#">Samples</a>
<ul>
<li><a href='father/father.html'>father</a></li>
<li><a href='mother/mother.html'>mother</a></li>
<li><a href='son/son.html'>son</a></li>
<span></span>
</ul>
            </li>
            <li><a href="pages/pipeline.html">Pipeline</a> </li>
          	<li><a href="pages/parameter.html">Parameter</a></li>
          	<li><a href="pages/help.html">Help</a></li>
            </ul>
        <br style="clear: left" />
    </div> <!-- end of menu -->
    
    <div id="templatemo_slider_wrapper">
      <div id="htmlcaption" class="nivo-html-caption">
       	  <strong>This</strong> is an example of a HTML caption with <a href="#">a link</a>.
        </div>
    
    </div>
    
    <script type="text/javascript" src="js/jquery-1.4.3.min.js"></script>
    <script type="text/javascript" src="js/jquery.nivo.slider.js"></script>
    <script type="text/javascript">
    $(window).load(function() {
    $('#slider').nivoSlider();
    });
    </script>
    
    <div id="templatemo_main"><br class="cleaner" />
<p>On IE, please ENABLE scripts or ActiveX.</p>
<h1>Analysis Summary</h1>
<p>3 sample(s) in this analysis: father mother son</br>Multi-sample variant calling ENABLED, assume all samples in the same breed.</br>Input is exome (or captured) sequencing data.</br>3 variant caller(s) used: GATK_HaploTyperCaller SAMtools VarScan</br>1 aligner(s) used: bwamem</br>File used for caculating coverage statistics and extracting variants: hg19_exome.bed</br>Readgroup : READGROUP_father,READGROUP_mother,READGROUP_son</br>Library : LIBRARY</br>Sequencing platform: ILLUMINA</br>Reference genome build is hg19</br>dbsnp138 is used for variant calling and recalibration (in GATK VQSR).</br>Java memory usage is limited to 1750m</br>Max number of processes: 8</br>Two INPUT files detected, PAIRED ends mode</br>Variant calling are run in parallel mode [EXPERIMENTAL].</br>###################################################</br>FINAL OUTPUT for father:</br>###################################################</br>NOTICE: Both consensus results and individual results are listed.</br>WARNING: consensus results may not be available because some variant callers or aligners may fail to generate output.</br>father_result/father.extract_consensus.vcf</br>father_result/father_bwamem.sort.rmdup.readfiltered.realn.recal_gatkhc.multi-call.filter.extract.vcf</br>father_result/father_bwamem.sort.rmdup.readfiltered_samtools.multi-call.filter.extract.vcf</br>father_result/father_bwamem.sort.rmdup.readfiltered_varscan.multi-call.filter.extract.vcf</br>###################################################</br>FINAL OUTPUT for mother:</br>###################################################</br>NOTICE: Both consensus results and individual results are listed.</br>WARNING: consensus results may not be available because some variant callers or aligners may fail to generate output.</br>mother_result/mother.extract_consensus.vcf</br>mother_result/mother_bwamem.sort.rmdup.readfiltered.realn.recal_gatkhc.multi-call.filter.extract.vcf</br>mother_result/mother_bwamem.sort.rmdup.readfiltered_samtools.multi-call.filter.extract.vcf</br>mother_result/mother_bwamem.sort.rmdup.readfiltered_varscan.multi-call.filter.extract.vcf</br>###################################################</br>FINAL OUTPUT for son:</br>###################################################</br>NOTICE: Both consensus results and individual results are listed.</br>WARNING: consensus results may not be available because some variant callers or aligners may fail to generate output.</br>son_result/son.extract_consensus.vcf</br>son_result/son_bwamem.sort.rmdup.readfiltered.realn.recal_gatkhc.multi-call.filter.extract.vcf</br>son_result/son_bwamem.sort.rmdup.readfiltered_samtools.multi-call.filter.extract.vcf</br>son_result/son_bwamem.sort.rmdup.readfiltered_varscan.multi-call.filter.extract.vcf</br></br></p>
<h1>Result highlights</h1>
<h2>father</h2>
<p>There are 81093904 QC-passed reads.</br>72469565(89.36%) reads are mapped. </br> 53.22% reads are mapped to target region.</br>Average coverage is  69.91 (in target region if this is captured sequencing).</br>29494 SNPs are called.</br>1690 indels are found.</br></p></br>
<h2>mother</h2>
<p>There are 71205444 QC-passed reads.</br>64073406(89.98%) reads are mapped. </br> 61.07% reads are mapped to target region.</br>Average coverage is  63.76 (in target region if this is captured sequencing).</br>29411 SNPs are called.</br>1610 indels are found.</br></p></br>
<h2>son</h2>
<p>There are 138779950 QC-passed reads.</br>124428082(89.66%) reads are mapped. </br> 52.49% reads are mapped to target region.</br>Average coverage is  113.44 (in target region if this is captured sequencing).</br>30157 SNPs are called.</br>1740 indels are found.</br></p></br>
<p></br><b>Details can be found on each SAMPLE page.</b></p>



    </div> <!-- end of main -->
    
    <div id="templatemo_footer">
      <div class="cleaner"></div>
    </div> <!-- end of footer -->
</div>

<div id="templatemo_cr_bar_wrapper">
	<div id="templatemo_cr_bar">
    	Copyright 2014 <a href="http://genomics.usc.edu">USC Wang Lab</a> | Designed by <a href="http://www.templatemo.com" target="_parent">Free CSS Templates</a>
    </div>
</div>


<script type='text/javascript' src='js/logging.js'></script>
</body>
</html>
